src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -374,6 +380,672 @@ static void appendModifiers(ImmutableArray<CustomModifier> customModifiers, Stri
+            {
+                appendIdentifier(namedArgument.Key, builder);
+                builder.Append(" = ");
+                appendAttributeArgument(namedArgument.Value, builder);
+            }
+
+            static void appendPrimitive(object? value, StringBuilder builder, bool withCast)
+            {
+                var options = ObjectDisplayOptions.UseQuotes | ObjectDisplayOptions.EscapeNonPrintableCharacters;
+                if (withCast)
+                {
+                    string cast = value switch
+                    {
+                        sbyte => "(sbyte)",
+                        short => "(short)",
+                        int => "(int)",
+                        long => "(long)",
+                        byte => "(byte)",
+                        ushort => "(ushort)",
+                        uint => "(uint)",
+                        ulong => "(ulong)",
+                        float => "(float)",
+                        double => "(double)",
+                        _ => "",
+                    };
+                    builder.Append(cast);
+                }
+
+                switch (value)
+                {
+                    case bool b:
+                        builder.Append(b ? "true" : "false");
+                        break;
+                    case sbyte sb:
+                        builder.Append(sb.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case short sh:
+                        builder.Append(sh.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case int i:
+                        builder.Append(i.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case long l:
+                        builder.Append(l.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case byte by:
+                        builder.Append(by.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case ushort us:
+                        builder.Append(us.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case uint ui:
+                        builder.Append(ui.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case ulong ul:
+                        builder.Append(ul.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case float f:
+                        // Note: we're printing as bits to avoid any loss
+                        builder.Append(BitConverter.ToInt32(BitConverter.GetBytes(f), startIndex: 0));
+                        break;
+                    case double d:
+                        // Note: we're printing as bits to avoid any loss
+                        builder.Append(BitConverter.DoubleToInt64Bits(d));
```

Comment:
>BitConverter.DoubleToInt64Bits(d) [](http://example.com/codeflow?start=39&length=33)

> The order of bits in the integer returned by the DoubleToInt64Bits method depends on whether the computer architecture is little-endian or big-endian.

This feels undesirable since the output will depend on computer architecture.